module/os/linux/zfs/zvol_os.c

```diff
@@ -40,18 +40,41 @@
 
 #include <linux/blkdev_compat.h>
 #include <linux/task_io_accounting_ops.h>
+#include <linux/blk-mq.h>
+
+static void zvol_request_impl(zvol_state_t *zv, struct bio *bio,
+    boolean_t force_sync);
 
 unsigned int zvol_major = ZVOL_MAJOR;
 unsigned int zvol_request_sync = 0;
 unsigned int zvol_prefetch_bytes = (128 * 1024);
 unsigned long zvol_max_discard_blocks = 16384;
-unsigned int zvol_threads = 32;
 unsigned int zvol_open_timeout_ms = 1000;
 
+/*
+ * zvol_threads is the module param the user passes in.
+ *
+ * zvol_actual_threads is what we use internally, since the user can pass
+ * zvol_thread = 0 to mean "use all the CPUs" (the default).  So on a quad
+ * core system, you would have: zvol_threads = 0, zvol_actual_threads = 4.
+ */
+unsigned int zvol_threads = 0;
+unsigned int zvol_actual_threads;
+
+#ifdef HAVE_BLK_MQ
+kmem_cache_t *blk_mq_cache = NULL;
+unsigned int zvol_blk_mq_queue_depth = BLKDEV_MAX_RQ;
```

Comment:
Reported by the kernel.org builder.

```
fs/zfs/zfs/../os/linux/zfs/zvol_os.c:66:40: error: ‘BLKDEV_MAX_RQ’ undeclared here (not in a function); did you mean ‘BLKDEV_MIN_RQ’?
 unsigned int zvol_blk_mq_queue_depth = BLKDEV_MAX_RQ;
                                        ^~~~~~~~~~~~~
                                        BLKDEV_MIN_RQ
fs/zfs/zfs/../os/linux/zfs/zvol_os.c: In function ‘zvol_os_create_minor’:
fs/zfs/zfs/../os/linux/zfs/zvol_os.c:1329:3: warning: ignoring return value of ‘add_disk’, declared with attribute warn_unused_result [-Wunused-result]
   add_disk(zv->zv_zso->zvo_disk);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   ```

Reply:
Sigh... looks like they renamed it to `BLKDEV_DEFAULT_RQ` in later kernels (https://lkml.org/lkml/2021/10/5/375).  I'll update it

Reply:
I fixed this in my latest push